src/app/dependencies/inversify.config.ts

```diff
@@ -0,0 +1,12 @@
+import {Container} from 'inversify'
+import {TYPES} from './types'
+import {PlaylistDao} from '../services/playlists/PlaylistDao'
+import {PlaylistService} from '../services/playlists/PlaylistService'
+import {UserDao} from '../services/users/UserDao'
+import {UserService} from '../services/users/UserService'
+
+export const container = new Container()
+container.bind<PlaylistDao>(TYPES.PlaylistDao).to(PlaylistDao)
+container.bind<PlaylistService>(TYPES.PlaylistService).to(PlaylistService)
+container.bind<UserDao>(TYPES.UserDao).to(UserDao)
+container.bind<UserService>(TYPES.UserService).to(UserService)
```

Comment:
Cool. I think we can actually take the Dao out of the service locator and just instantiate them in the services, but can do that opportunistically

Reply:
In the [Inversify Docs](https://github.com/inversify/InversifyJS) they are against having a hard dependency on another class, and only want one point in the [root of your app](http://blog.ploeh.dk/2011/07/28/CompositionRoot/) where the actual objects are injected.

They're also pretty harsh on [Service Locator](http://blog.ploeh.dk/2010/02/03/ServiceLocatorisanAnti-Pattern/).

Frankly I don't do any DI stuff so I don't have any opinions one way or another, I'll defer to what you think.

Reply:
Gonna mess around with this tonight (and also try to get you that IAM user).